a10_octavia/controller/worker/tasks/a10_database_tasks.py

```diff
@@ -690,13 +639,29 @@ def execute(self, member_list):
             for member in member_list:
                 if member.subnet_id not in member_subnet:
                     pool_count_subnet = self.member_repo.get_pool_count_subnet(
-                        db_apis.get_session(), member.project_id, member.subnet_id)
+                        db_apis.get_session(), partition_project_list, member.subnet_id)
                     lb_count_subnet = self.loadbalancer_repo.get_lb_count_by_subnet(
-                        db_apis.get_session(), member.project_id, member.subnet_id)
+                        db_apis.get_session(), partition_project_list, member.subnet_id)
                     if pool_count_subnet <= 1 and lb_count_subnet == 0:
                         subnet_list.append(member.subnet_id)
                     member_subnet.append(member.subnet_id)
             return subnet_list
         except Exception as e:
             LOG.exception("Failed to get subnet list for members: %s", str(e))
             raise e
+
+
+class GetProjectsForPartition(BaseDatabaseTask):
```

Comment:
The naming on this needs to better convey that it's meant for child projects

ie `GetChildProjectsOfParentPartition`

Reply:
Here we are not checking multitenancy settings, hence it will be used for all the projects no matter whether it is a parent or children.

Reply:
It's about intention though. We never want multiple projects to be tied to one partition. The only time this is okay is if these are child projects.

So we intend this class to be used in that way and should name it as such

Reply:
0k will change the function name.